### PR TITLE
Add prefix and infix operators

### PR DESCRIPTION
**Pull request by AstroSnail**
progress tracking pr

### CODE REVIEW

**Comment by AstroSnail** [comment]
merged some other fixes from #90 because i was annoyed at not having them here

**Review by aiverson — APPROVED** [review]

File: evaluator.lua [hunk 1]
```diff
@@ -339,9 +339,11 @@ local check_concrete
 -- indexed by kind x kind
 local concrete_comparers = {}
 
+---@alias value_comparer fun(a: value, b: value, typechecker: TypeCheckerState): boolean, string?
```
Comment (LunNova) [inline evaluator.lua:342]: yay

File: base-env.lua [hunk 2]
```diff
@@ -38,7 +38,7 @@ local val_desc_empty = value.enum_value("empty", val_tup_cons())
 --- lua_operative is dependently typed and should produce inferrable vs checkable depending on the goal, and an error as the second return if it failed
 --- | unknown in the second return insufficiently constrains the non-error cases to be inferrable or checkable terms
 --- this can be fixed in the future if we swap to returning a Result type that can express the success/error constraint separately
----@alias lua_operative fun(syntax : ConstructedSyntax, env : Environment, goal : expression_goal) : boolean, inferrable | checkable | unknown, Environment?
+---@alias lua_operative fun(syntax : ConstructedSyntax, env : Environment, goal : expression_goal) : boolean, inferrable | checkable, Environment?
```
Comment (LunNova) [inline base-env.lua:41]: isn't this making yellow squigglies for any error case

Comment (LunNova) [reply base-env.lua:41]: if you managed to make it not update the comments right above it

**Pull request merged**